### PR TITLE
Update system.h for other users of ex2_obc_software

### PR DESCRIPTION
**Pull request by tganley**
Fix the previously-unaccounted-for fact that communication protocol numbers do not correspond to MCU register numbers, for instance UART1 is actually sciREG4.

### CODE REVIEW

File: main/system.h [hunk 1]
```diff
@@ -55,21 +48,22 @@
...
 #else // These values are expected to be adjusted based on what the developer is working on
-#define CSP_SCI sciREG4
-#define GPS_SCI sciREG2
-#define ADCS_SCI sciREG3
-#define PAYLOAD_SCI 0
-#define DFGM_SCI 0
+    #define CSP_SCI sciREG2
+    #define GPS_SCI sciREG1
+    #define ADCS_SCI sciREG3
+    #define PAYLOAD_SCI sciREG1
```
Comment (Taylorr82) [inline main/system.h:64]: having multiple definitions of the same sciREG will cause issues in the interrupt switch() statements. We need to come up with a way to exclude compiling certain things based on the payload installed. Probably similar to IS_ATHENA

File: main/system.h [hunk 2]
```diff
@@ -23,18 +23,11 @@
 #include "FreeRTOS.h"
 #include <stdint.h>
 
-#define config_IS_ATHENA 0
-#define config_IS_FLATSAT 0
+#ifdef IS_FLATSAT
+    #ifndef IS_ATHENA
+    #error If IS_FLATSAT is defined then IS_ATHENA must be defined
+    #endif
```
Comment (Taylorr82) [inline main/system.h:29]: much more concise than the way I did it

File: main/system.h [hunk 3]
```diff
@@ -83,15 +77,17 @@
...
+    #define SD_SPI spiREG1          //?
 #else // These values are expected to be adjusted based on what the developer is working on
-#define IRIS_CONFIG_SPI spiREG1
-#define IRIS_SPI spiREG4
-#define SBAND_SPI spiREG2
-#define SD_SPI spiREG3
+    #define IRIS_CONFIG_SPI spiREG4 //SPI1
+    #define IRIS_SPI spiREG5        //SPI3
+    #define SBAND_SPI spiREG3       //SPI2
+    #define SD_SPI spiREG1          //?
```
Comment (Taylorr82) [inline main/system.h:89]: Did you change these knowing they are likely to get all messed up by development? I don't think that's necessarily a bad thing

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

**Comment by tganley** [comment]
This pull request is stale and has a lot of commits that were not intended to be merged. The changes to system.h will be carried into master once we're ready to merge everything.

**Pull request closed without merging**